.gitignore

```diff
@@ -0,0 +1,13 @@
+CMakeLists.txt.user
+CMakeCache.txt
```

Comment:
I recommend to not ignore the CMake generated files which are at the same level as the source files. This can encourage in-source build which is not a good practice. If an in-source build happens by accident, users could easily see which files have been generated (with git status) and remove them. But this won't be possible if the files are ignored.